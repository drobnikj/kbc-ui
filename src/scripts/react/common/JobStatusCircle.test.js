import React from 'react';

import JobStatusCircle from './JobStatusCircle';

jest.mock('@keboola/indigo-ui/src/kbc-bootstrap/img/status-green.svg', () => '/status-green.svg');
jest.mock('@keboola/indigo-ui/src/kbc-bootstrap/img/status-grey.svg', () => '/status-grey.svg');
jest.mock('@keboola/indigo-ui/src/kbc-bootstrap/img/status-orange.svg', () => '/status-orange.svg');
jest.mock('@keboola/indigo-ui/src/kbc-bootstrap/img/status-red.svg', () => '/status-red.svg');

jest.mock('../../stores/ApplicationStore', () => {
  return {
    getScriptsBasePath: () => {
      return '/some-path';
    }
  };
});

describe('<JobStatusCircle />', function() {
  it('should render with no props (grey)', function() {
    shallowSnapshot(
      <JobStatusCircle />
    );
  });

  it('should render with status success (green)', function() {
    shallowSnapshot(
      <JobStatusCircle status="success" />
    );
  });

  it('should render with status processing (orange)', function() {
    shallowSnapshot(
      <JobStatusCircle status="processing" />
    );
  });

  it('should render with status error (red)', function() {
    shallowSnapshot(
      <JobStatusCircle status="error" />
    );
  });

  it('should render with status warn (red)', function() {
    shallowSnapshot(
      <JobStatusCircle status="warn" />
    );
  });

  it('should render with status warning (red)', function() {
    shallowSnapshot(
      <JobStatusCircle status="warning" />
    );
  });
});
