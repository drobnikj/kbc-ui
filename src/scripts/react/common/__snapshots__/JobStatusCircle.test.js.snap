exports[`<JobStatusCircle /> should render with no props (grey) 1`] = `
<img
  src="/some-path/status-grey.svg" />
`;

exports[`<JobStatusCircle /> should render with status error (red) 1`] = `
<img
  src="/some-path/status-red.svg" />
`;

exports[`<JobStatusCircle /> should render with status processing (orange) 1`] = `
<img
  src="/some-path/status-orange.svg" />
`;

exports[`<JobStatusCircle /> should render with status success (green) 1`] = `
<img
  src="/some-path/status-green.svg" />
`;

exports[`<JobStatusCircle /> should render with status warn (red) 1`] = `
<img
  src="/some-path/status-red.svg" />
`;

exports[`<JobStatusCircle /> should render with status warning (red) 1`] = `
<img
  src="/some-path/status-red.svg" />
`;
